src/ngx_http_lua_socket_tcp.c

```diff
@@ -6395,4 +6392,112 @@ ngx_http_lua_ffi_socket_tcp_del_udata(ngx_http_lua_socket_tcp_upstream_t *u,
 }
 
 
+int
+ngx_http_lua_ffi_socket_tcp_getoption(ngx_http_lua_socket_tcp_upstream_t *u,
+    int option, int *val, u_char *err, size_t *errlen)
+{
+    u_char    errstr[NGX_MAX_ERROR_STR];
+    socklen_t len;
+    u_char   *p;
+    int       fd, r;
+
+    if (u == NULL || u->peer.connection == NULL) {
+        *errlen = ngx_snprintf(err, *errlen, "closed") - err;
+        return NGX_ERROR;
+    }
+
+    fd = u->peer.connection->fd;
+    len = sizeof(int);
+
+    switch (option) {
+    case OPTION_NAME_KEEPALIVE:
+        r = getsockopt(fd, SOL_SOCKET, SO_KEEPALIVE, (void *) val, &len);
+        break;
+
+    case OPTION_NAME_REUSEADDR:
+        r = getsockopt(fd, SOL_SOCKET, SO_REUSEADDR, (void *) val, &len);
+        break;
+
+    case OPTION_NAME_TCP_NODELAY:
+        r = getsockopt(fd, IPPROTO_TCP, TCP_NODELAY, (void *) val, &len);
+        break;
+
+    case OPTION_NAME_SNDBUF:
+        r = getsockopt(fd, SOL_SOCKET, SO_RCVBUF, (void *) val, &len);
+        break;
+
+    case OPTION_NAME_RCVBUF:
+        r = getsockopt(fd, SOL_SOCKET, SO_SNDBUF, (void *) val, &len);
+        break;
+
+    default:
+        *errlen = ngx_snprintf(err, *errlen, "unsupported option") - err;
```

Comment:
We should output the unknown option value in the error message to make debugging easier.

Reply:
accept

Reply:
@poorsea You still haven't changed it.

Reply:
sorry， miss it

Reply:
miss it, check and done